[WIP] OBSDOCS-1079: Create a table listing and describing all RBAC permissi… (#80127)
openshift/openshift-docs: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc

* To monitor user-defined projects.
* To configure the components that monitor user-defined projects.
* To configure alert routing for user-defined projects.

> * To monitor or manage alerts and silences for user-defined projects.

Reviewer: ```suggestion
* To manage alerts and silences for user-defined projects.
```

You can grant the permissions by assigning one of the following monitoring roles and cluster roles:

.Monitoring roles
|===
|Role name |Description

> |`user-workload-monitoring-config-edit` |This role is given in the `openshift-user-workload-monitoring` project. Users with this role can edit the `user-workload-monitoring-config` `ConfigMap` object to configure Prometheus, Prometheus Operator, Alertmanager, and Thanos Ruler for user-defined workload monitoring.

Reviewer: not sure about "This role is given ...".

> |`monitoring-alertmanager-api-reader` |Users with this role have read access to the Alertmanager API for monitoring of user-defined projects. They can also view alert silences in the *Developer* perspective of the {product-title} web console.

Reviewer: ```suggestion
|`monitoring-alertmanager-api-reader` |Users with this role have read access to the user-defined Alertmanager API (if enabled). They can also view alert silences in the *Developer* perspective of the {product-title} web console.
```

|`monitoring-alertmanager-api-writer` |Users with this role have read and write access to the Alertmanager API for monitoring of user-defined projects. They can also manage alert silences in the *Developer* perspective of the {product-title} web console.
|===

.Monitoring cluster roles
|===
|Cluster role name |Description

> |`monitoring-rules-view` |Users with this cluster role have read access to `PrometheusRule` custom resources (CRs) for user-defined projects. They can also view the alerts in the *Developer* perspective of the {product-title} web console.

Reviewer: the cluster role can be granted for specific project(s) with RoleBinding or for all projects with ClusterRoleBinding (but the latter is probably not recommended for security concerns).

Author: It the latter is not recommended, I would not mention it in our docs. It is better to include only the recommended way

Author: This actually leads me to this question: https://github.com/openshift/openshift-docs/pull/80127#issuecomment-2288770100

> |`monitoring-rules-edit` |Users with this cluster role can create, modify, and delete `PrometheusRule` CRs for user-defined projects. They can also create and silence alerts in the *Developer* perspective of the {product-title} web console.

Reviewer: ```suggestion
|`monitoring-rules-edit` |Users with this cluster role can create, modify, and delete `PrometheusRule` CRs for user-defined projects. They can also manage alert silences in the *Developer* perspective of the {product-title} web console.
```

|`monitoring-edit` |Users with this cluster role have the same privileges as users with the `monitoring-rules-edit` cluster role. Additionally, users can create, modify, and delete `ServiceMonitor` and `PodMonitor` resources to scrape metrics from services and pods.

> |`alert-routing-edit` |Users with this cluster role can create, update, and delete `AlertmanagerConfig` CRs for user-defined projects.

Reviewer: ```suggestion
|`alert-routing-edit` |Users with this cluster role can create, update, and delete `AlertmanagerConfig` custom resources for user-defined projects.
```

Author: Because I already used "custom resources (CRs)" for the `monitoring-rules-view` row, based on the writing rules I should only mention CRs from now on. D you think it can be problematic here? Or it is okay if I keep it as it is?

* To configure the components that monitor user-defined projects.
* To configure alert routing for user-defined projects.
* To manage alerts and silences for user-defined projects.

You can grant the permissions by assigning one of the following monitoring roles and cluster roles:

[IMPORTANT]
====
Monitoring cluster roles can be bound with `RoleBinding` to any user project. The monitoring roles can be only bound to the `openshift-user-workload-monitoring` project.
====

.Monitoring roles
|===
|Role name |Description

|`user-workload-monitoring-config-edit` 
|Users with this role can edit the `user-workload-monitoring-config` `ConfigMap` object to configure Prometheus, Prometheus Operator, Alertmanager, and Thanos Ruler for user-defined workload monitoring. 

|`monitoring-alertmanager-api-reader` 
|Users with this role have read access to the user-defined Alertmanager API for all projects, if the user-defined Alertmanager is enabled.

|`monitoring-alertmanager-api-writer` 
|Users with this role have read and write access to the user-defined Alertmanager API for all projects, if the user-defined Alertmanager is enabled.
|===

.Monitoring cluster roles
|===

> |Cluster role name |Description

Reviewer: (discussion) the other table for core monitoring has a `project` column. Should have it here too?

Author: I added a project column to the other one so that it differentiates between the roles and that one cluster role. I could do the same here, but it would be just a lot of repetition I think :frowning_face:  That is why, instead, I decided to add the warning: 	
Monitoring cluster roles can be bound with RoleBinding to any user project. The monitoring roles can be only bound to the openshift-user-workload-monitoring project.

But if you think it would be better to add that column, I can :slightly_smiling_face: maybe that would be clearer :+1:

Reviewer: That's fine by me. I just wanted to make sure that we were good with it :)

|`monitoring-rules-view` 
|Users with this cluster role have read access to `PrometheusRule` custom resources (CRs) for user-defined projects. They can also view the alerts and silences in the *Developer* perspective of the {product-title} web console.

|`monitoring-rules-edit` 
|Users with this cluster role can create, modify, and delete `PrometheusRule` CRs for user-defined projects. They can also manage alerts and silences in the *Developer* perspective of the {product-title} web console.

|`monitoring-edit` 

> |Users with this cluster role have the same privileges as users with the `monitoring-rules-edit` cluster role. Additionally, users can create, modify, and delete `ServiceMonitor` and `PodMonitor` resources to scrape metrics from services and pods.

Reviewer: (nit)
```suggestion
|Users with this cluster role have the same privileges as users with the `monitoring-rules-edit` cluster role. Additionally, users can create, read, modify, and delete `ServiceMonitor` and `PodMonitor` resources to scrape metrics from services and pods.
```

* To monitor user-defined projects.
* To configure the components that monitor user-defined projects.
* To configure alert routing for user-defined projects.

> * To manage alerts and silences for user-defined projects.

Reviewer 2: I know this is outside the scope of this PR, but IBM style guide suggests something like:

```
You can also grant developers and other users and of the following permissions:
* Monitoring user-defined projects
* Configuring the components that monitor user-defined projects
* Configuring alert routing for user-defined projects
* Managing alerts and silences for user-defined projects
```

> You can grant the permissions by assigning one of the following monitoring roles and cluster roles:

Reviewer 2: Should this be _or_, as the user selects only one?
 ```suggestion
You can grant the permissions by assigning one of the following monitoring roles or cluster roles:
```

Author: Hello @mburke5678 I changed it to "or" as you suggested. Is everything else okay (I see that `peer-review-done` label is not applied yet. Thanks!
